src/OrbitGl/ScopeTree.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
```

Comment:
General question regarding the implementation:
There are a couple of recursive functions, that write the results into a parameter rather than returning them, is this because auf tail-call optimization? Have you checked if those functions get tail-call optimized? I'd assume that should be critical for performance. 
If so, maybe add a comment.